### PR TITLE
Task/dummy web server additional request hendlers dima

### PR DESCRIPTION
**Pull request by Promah**

### CODE REVIEW

**Comment by Promah** [comment]
definitely have problems on  "/api/quiz/submit/0"

**Comment by Promah** [comment]
> * please rework to use GSON  it will simplifies parsing to one line
> * please move json responses from constant string to resources (just put in same class path as mock server code) and load them using `getClass().getResourceAsStream("quizList.json")`

- reworked to use GSON
- json responses from constant string moved to resources

File: app/src/main/java/ua/onseo/ds/mock_server/DummyWebService.kt [hunk 1]
```diff
@@ -30,14 +31,24 @@ class DummyWebService : NanoHTTPD(8080) {
...
+            "/api/quiz/submit/0" -> handleQuizSubmit0(session)
             else -> handleOther()
         }
     }
 
     private fun handleSignIn(session: IHTTPSession): Response {
         val parameters: Map<String, List<String>> = session.parameters
         return if (parameters.containsKey("username") && parameters.containsKey("password")) {
-            val msg = "{\"status\":\"success\", \"token\":\"$HEADER_VALUE\"}"
+            val msg = "{" +
```
Comment (YPoudanenOnSeo) [inline app/src/main/java/ua/onseo/ds/mock_server/DummyWebService.kt]: try to use kotlin multiline string

File: app/src/main/java/ua/onseo/ds/mock_server/DummyWebService.kt [hunk 2]
```diff
@@ -30,14 +31,24 @@ class DummyWebService : NanoHTTPD(8080) {
...
+                    "\"accessToken\":\"$HEADER_VALUE\", " +
```
Comment (YPoudanenOnSeo) [inline app/src/main/java/ua/onseo/ds/mock_server/DummyWebService.kt]: try to use kotlin multiline string

File: gradlew.bat [hunk 3]
```diff
@@ -1,84 +1,84 @@
...
-rem Set variable GRADLE_EXIT_CONSOLE if you need the _script_ return code instead of
-rem the _cmd.exe /c_ return code!
-if  not "" == "%GRADLE_EXIT_CONSOLE%" exit 1
-exit /b 1
-
-:mainEnd
-if "%OS%"=="Windows_NT" endlocal
-
-:omega
+@if "%DEBUG%" == "" @echo off
```
[the remark this answers is not in the document]
Comment (Promah) [reply gradlew.bat]: Really not known why file is committed, file contents are the same.
image

File: app/src/main/java/ua/onseo/ds/repository/api/JsonParser.kt [hunk 4]
```diff
@@ -0,0 +1,160 @@
+package ua.onseo.ds.repository.api
+
+import com.google.gson.Gson
+import org.json.JSONObject
+import ua.onseo.ds.repository.model.ServerErrors
+import ua.onseo.ds.repository.model.ServerResponse
+
+object JsonParser {
```
[the remark this answers is not in the document]
Comment (Promah) [reply app/src/main/java/ua/onseo/ds/repository/api/JsonParser.kt]: JsonParser.kt deleted, his functional implemented in new generic sendRequest function

**Comment by Promah** [comment]
continuation in branch "task/login_view_model_dima"

**Pull request closed without merging**